internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -162,13 +162,13 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
         case _ => newOne()
       }
     }
-    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = {
+    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = if (dep != NoSymbol) {
```

Comment:
We should also guard against `null`, just to be safer since scalac can unexpectedly give us a null from time to time. Perhaps it's a good idea to create a utility called `ignoreSymbol` that will guard against `NoSymbol` and `null` at the same time? It's a very recurrent logic in Zinc.